internal/bufferedread/buffered_reader.go

```diff
@@ -103,6 +103,61 @@ func NewBufferedReader(object *gcs.MinObject, bucket gcs.Bucket, config *Buffere
 	return reader, nil
 }
 
+func (p *BufferedReader) prefetch() error {
+	// Do not schedule more blocks if the prefetch queue has reached its maximum configured size.
+	availableSlots := p.config.MaxPrefetchBlockCnt - int64(p.blockQueue.Len())
+	if availableSlots <= 0 {
+		return nil
+	}
+	blockCountToPrefetch := min(p.numPrefetchBlocks, availableSlots)
+	if blockCountToPrefetch <= 0 {
+		return nil
+	}
+
+	logger.Debugf("Prefetching %d blocks", blockCountToPrefetch)
+
+	for i := int64(0); i < blockCountToPrefetch; i++ {
+		if p.nextBlockIndexToPrefetch >= p.totalBlockCount() {
```

Comment:
Can we compute the totalBlockCount once at the time of creation of buffered-reader, instead of computing this every time?